microsoft/testsuites/power/common.py

```diff
@@ -150,3 +171,34 @@ def cleanup_env(environment: Environment) -> None:
         kill.by_name("iperf3")
         kill.by_name("fio")
         kill.by_name("stress-ng")
+
+
+def _expand_os_partition(node: Node, log: Logger) -> None:
+    if isinstance(node.os, Redhat):
+        pv_result = node.execute("pvscan -s", sudo=True, shell=True).stdout
+        # The output of pvscan -s is like below.:
+        #  /dev/sda4
+        #  Total: 1 [299.31 GiB] / in use: 1 [299.31 GiB] / in no VG: 0 [0   ]
+        pattern = re.compile(r"(?P<disk>.*)(?P<number>[\d]+)", re.M)
+        matched = find_groups_in_lines(pv_result, pattern)
+        if not matched:
+            log.debug("LVM is not enabled on the VM. Does not require LV resize.")
+            return
+        disk = matched[0].get("disk")
+        number = matched[0].get("number")
+        node.execute(f"growpart {disk} {number}", sudo=True)
+        node.execute(f"pvresize {pv_result.splitlines()[0]}", sudo=True)
+        root_partition = node.tools[Mount].get_partition_info("/")[0]
+        device_name = root_partition.name
+        device_type = root_partition.type
+        cmd_result = node.execute(f"lvdisplay {device_name}", sudo=True)
+        if cmd_result.exit_code == 0:
```

Comment:
what happened, if the exit_code is not 0? Add logic or comments.

Reply:
Updated